### PR TITLE
feat(ci): enable GH-contained auto-updates

### PR DESCRIPTION
**Pull request by cjdcordeiro**
- [x] I have signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] I am following the [CONTRIBUTING](https://github.com/canonical/oci-factory/blob/main/CONTRIBUTING.md) guidelines

*Ping the @canonical/rocks team.*

---

## Description
Add standalone script to run auto updates for images, based on new Ubuntu container bases.

### CODE REVIEW

File: src/workflow-engine/charms/temporal-worker/oci_factory/activities/find_images_to_update.py [hunk 1]
```diff
@@ -122,12 +121,17 @@ def find_released_revisions(releases_json: dict) -> dict:
         uber_img_trigger = {"version": 1, "upload": []}
         # We'll also need to find which tags (channels) to release the new
         # rebuilds to
-        docker_tags_url = f"https://hub.docker.com/v2/repositories/%%%/{image}/tags"
+        # TODO: Get rid of this once we have a proper DB where to store all the
+        # image information.
+        # This is a bit nasty as these APIs return paginated results
+        # and don't offer enough querying parameters to filter the results.
+        ecr_tags_url = "https://api.us-east-1.gallery.ecr.aws/describeImageTags"
```
Comment (SamirPS) [inline src/workflow-engine/charms/temporal-worker/oci_factory/activities/find_images_to_update.py:128]: why you don't use boto3?

File: src/workflow-engine/charms/temporal-worker/oci_factory/activities/find_images_to_update.py [hunk 2]
```diff
@@ -157,35 +161,78 @@ def find_released_revisions(releases_json: dict) -> dict:
...
+                "external_ref_id": f"workflow-engine-{image}-{int(time.time())}",
+            },
+        }
+        wf_dispatch_url = str(
+            "https://api.github.com/repos/"
+            "canonical/oci-factory/"
+            "actions/workflows/Image.yaml/dispatches"
+        )
+
+        # TODO: remove test condition
```
Comment (SamirPS) [inline src/workflow-engine/charms/temporal-worker/oci_factory/activities/find_images_to_update.py:229]: why this?

**Review by SamirPS — CHANGES_REQUESTED** [review]

File: src/workflow-engine/charms/temporal-worker/oci_factory/activities/find_images_to_update.py (hunk 1 above)
Comment (cjdcordeiro) [reply src/workflow-engine/charms/temporal-worker/oci_factory/activities/find_images_to_update.py:128]: why the extra requirement?

Comment (cjdcordeiro) [reply src/workflow-engine/charms/temporal-worker/oci_factory/activities/find_images_to_update.py:128]: I'll make the changes in a new PR if you think there's value in using boto3. For now, I need to test this so I'll go ahead and merge.

**Pull request merged**